prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/channel/ChannelAddCommand.java

```diff
@@ -20,61 +20,95 @@
 import org.wildfly.channel.Channel;
 import org.wildfly.channel.ChannelManifestCoordinate;
 import org.wildfly.channel.Repository;
+import org.wildfly.channel.maven.ChannelCoordinate;
 import org.wildfly.prospero.actions.MetadataAction;
 import org.wildfly.prospero.api.ArtifactUtils;
+import org.wildfly.prospero.api.MavenOptions;
+import org.wildfly.prospero.api.ProvisioningDefinition;
 import org.wildfly.prospero.api.RepositoryUtils;
-import org.wildfly.prospero.cli.ActionFactory;
+import org.wildfly.prospero.api.TemporaryFilesManager;
+
+import org.wildfly.prospero.cli.ChannelUtils;
+import org.wildfly.prospero.cli.ArgumentParsingException;
 import org.wildfly.prospero.cli.CliConsole;
+import org.wildfly.prospero.cli.ActionFactory;
 import org.wildfly.prospero.cli.CliMessages;
 import org.wildfly.prospero.cli.RepositoryDefinition;
 import org.wildfly.prospero.cli.ReturnCodes;
-import org.wildfly.prospero.api.TemporaryFilesManager;
 import org.wildfly.prospero.cli.commands.AbstractCommand;
 import org.wildfly.prospero.cli.commands.CliConstants;
 import picocli.CommandLine;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
 import java.nio.file.Path;
 import java.util.List;
 import java.util.Optional;
 
 @CommandLine.Command(name = CliConstants.Commands.ADD, sortOptions = false)
 public class ChannelAddCommand extends AbstractCommand {
 
-    @CommandLine.Option(names = CliConstants.CHANNEL_NAME, required = true)
+    @CommandLine.Option(names = CliConstants.CHANNEL_NAME)
     private String channelName;
 
-    @CommandLine.Option(names = CliConstants.CHANNEL_MANIFEST, required = true)
-    private String gavUrlOrPath;
+    @CommandLine.Option(names = CliConstants.CHANNEL_MANIFEST)
+    private String manifestLocation;
```

Comment:
`channelName` should always be requires, even if the channelFile is provided. 

Also the command syntax should enforce that either `channelName`, `repositories` and `manifestLocation` is present or `channelLocation` is present. I thing grouping the arguments using `@CommandLine.ArgGroup` should be able to achieve that.

Reply:
Do you mean all three channelName, repo and manifestLocation should be present? If channelLocation is not present

Reply:
yes, exactly